web/packages/shared/components/Notification/Notification.tsx

```diff
@@ -237,9 +239,20 @@ const NotificationBody = ({
           {content.description}
         </Text>
       )}
-      {content.action && (
+      {action && (
         <Box alignSelf="flex-start">
-          <ActionButton intent="neutral" action={content.action} />
+          <ActionButton
+            intent="neutral"
+            action={{
+              href: action.href,
+              content: action.content,
+              onClick: event => {
+                // Prevents toggling the isExpanded flag.
+                event.stopPropagation();
```

Comment:
Is there any way we can add a test for this? It's a rather niche interaction that can easily regress in the future. OTOH, as `isExpanded` merely controls the styles, I don't think the JSDOM-based tests are going to be able to detect the difference.

Reply:
Yeah, we can't detect the text overflow. Instead, I added a test that checks if the style wasn't modified by the click. What you think?

Reply:
That should be good enough for now! I didn't realize that `toHaveStyleRule` works with CSS in general, I've always assumed it only works when the style is set directly on a node.